simdev/module/git/clone_stage.py

```diff
@@ -0,0 +1,140 @@
+import logging
+import os
+import textwrap
+
+from git import GitCommandError
+from pydriller import Repository
+from tqdm import tqdm
+
+from simdev.util.pipeline import Pipeline, Stage, PipelineException
+
+
+class AuthorCompound:
+    """
+    A simple structure that embodies both author's email and name
+    """
+
+    def __init__(self, name: str, email: str):
+        self.name = name
+        self.email = email
+
+    def __repr__(self) -> str:
+        """
+        :return: info about a contributor
+        """
+        return "%s <%s>" % (self.name, self.email)
+
+    def __hash__(self) -> int:
+        return hash((self.name, self.email))
```

Comment:
it's not reproducible among different runs - [link]((https://stackoverflow.com/questions/27522626/hash-function-in-python-3-3-returns-different-results-between-sessions) - I would recommend to use hashlib.sha256